src/bigunit.ts

```diff
@@ -555,13 +634,26 @@ export class BigUnit {
     const fractionalPartBigInt = BigInt(paddedFractionalPart);
 
     // Combine the integer and fractional parts
-    const combinedValueBigInt =
-      integerPartBigInt * BigInt(10 ** precision) + fractionalPartBigInt;
+    let combinedValueBigInt =
+      integerPartBigInt * precisionFactor + fractionalPartBigInt;
+
+    // If the integer part is negative, convert the fractional part to a negative number otherwise it will incorrectly be positive
+    if (integerPart.includes("-")) {
```

Comment:
It can only be the first char. We we should just check integerPart[0] which will save a lot of computation.